controller/src/main/java/org/jboss/as/controller/ParallelBootOperationStepHandler.java

```diff
@@ -358,9 +366,10 @@ public void rollback() {}
                     }, new ModelNode());
                     return;
                 }
+
                 operationContext = new ParallelBootOperationContext(transactionControl, processState,
                         primaryContext, runtimeOps, controllingThread, controller, lockId, controller.getAuditLogger(),
-                        controller.getManagementModel().getRootResource(), extraValidationStepHandler);
+                        controller.getManagementModel().getRootResource(), extraValidationStepHandler, bootSecurityDomain::getAnonymousSecurityIdentity);
```

Comment:
The problem we have at the moment is running in legacy mode and running in fully configured mode we have different definitions of where the security domain is located and different ways locate it.  

On boot regardless of which domain you use the identity would always be anonymous.

This may be easier to centralise once we can pull out the legacy realms.